src/configs/ManageProducts.config.js

```diff
@@ -95,8 +96,11 @@ const ManageProducts = (plugins) => [
       isDisabled: () => plugins.status?.woocommerce !== "Active",
     },
     actions: {
-      buttonClick: (state, setShowModal) => {
-        window.location.href = getUrl("post-new.php?post_type=gift_card");
+      buttonClick: async (state, setShowModal) => {
+        await createProduct({ type: "gift-card", status: "draft" }).then(
```

Comment:
Since we're using `async`/`await` we don't need to chain `.then`

```js
let product = await createProduct({ type: "gift-card", status: "draft" });
window.location.href = `post.php?post=${product?.id}&action=edit`
```